File: src/main/java/hudson/plugins/accurev/parsers/xml/ParseFiles.java
package hudson.plugins.accurev.parsers.xml;

import static jenkins.plugins.accurev.util.AccurevUtils.cleanAccurevPath;

import java.io.IOException;
import java.util.List;

import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserException;

import hudson.plugins.accurev.AccurevElement;
import hudson.plugins.accurev.AccurevLauncher;

public class ParseFiles implements AccurevLauncher.ICmdOutputXmlParser<Boolean, List<AccurevElement>> {

    @Override
    public Boolean parse(XmlPullParser parser, List<AccurevElement> context) throws AccurevLauncher.UnhandledAccurevCommandOutput, IOException, XmlPullParserException {
        while (parser.next() != XmlPullParser.END_DOCUMENT) {
            if (parser.getEventType() == XmlPullParser.START_TAG) {
                if ("element".equals(parser.getName())) {
                    String location = cleanAccurevPath(parser.getAttributeValue("", "location"));
                    String status = parser.getAttributeValue("", "status");
                    context.add(new AccurevElement(location, status));
                }
            }
        }
        return context != null;
    }
}

File: src/main/kotlin/hudson/plugins/accurev/parsers/xml/ParseFiles.kt
package hudson.plugins.accurev.parsers.xml

import hudson.plugins.accurev.AccurevElement
import hudson.plugins.accurev.AccurevLauncher
import hudson.plugins.accurev.AccurevUtils.Companion.cleanAccurevPath
import org.xmlpull.v1.XmlPullParser

class ParseFiles : AccurevLauncher.ICmdOutputXmlParser<Boolean, ArrayList<AccurevElement>> {
    override fun parse(parser: XmlPullParser?, context: ArrayList<AccurevElement>?): Boolean {
        while (parser?.next() != XmlPullParser.END_DOCUMENT) {
            if (parser?.eventType == XmlPullParser.START_TAG) {
                if ("element".equals(parser.name, ignoreCase = true)) {
                    val location = cleanAccurevPath(parser.getAttributeValue("", "location"))
                    val status = parser.getAttributeValue("", "status")
                    context?.add(AccurevElement(location, status))
                }
            }
        }
        return context != null
    }
}
